### PR TITLE
Throwing into disposals logic splitting

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
The “insert items in container on throw” behavior should not be tied only to the disposals system
is now a separate component that can easily be assigned to any entity with ContainerContainer

no game changes
no cl

### CODE REVIEW

File: Content.Server/Containers/ThrowInsertContainerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/// <summary>
+/// Allows objects to fall inside the Container when thrown
+/// </summary>
+[RegisterComponent]
+[Access(typeof(ThrowInsertContainerSystem))]
+public sealed partial class ThrowInsertContainerComponent : Component
+{
+    [DataField(required: true)]
+    public string? ContainerId;
```
Comment (deltanedas) [inline Content.Server/Containers/ThrowInsertContainerComponent.cs:13]: no reason for this to be null, `string ContainerId = string.Empty` and remove null checks

File: Content.Server/Containers/ThrowInsertContainerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+using Robust.Shared.Containers;
+using Robust.Shared.Random;
+
+namespace Content.Server.Containers;
+
+public sealed class ThrowInsertContainerSystem : EntitySystem
+{
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
```
Comment (deltanedas) [inline Content.Server/Containers/ThrowInsertContainerSystem.cs:15]: _container

File: Content.Server/Containers/ThrowInsertContainerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            return;
+
+        var container = _containerSystem.GetContainer(ent, ent.Comp.ContainerId);
+
+        if (!_containerSystem.CanInsert(args.Thrown, container))
+            return;
+
+
+        var rand = _random.NextFloat();
+        if (rand > ent.Comp.Probability)
```
Comment (deltanedas) [inline Content.Server/Containers/ThrowInsertContainerSystem.cs:38]: _random.Prob

File: Content.Server/Containers/ThrowInsertContainerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            _audio.PlayPvs(ent.Comp.MissSound, ent);
+            _popup.PopupEntity(Loc.GetString(ent.Comp.MissLocString), ent);
+            return;
+        }
+
+        if (_containerSystem.Insert(args.Thrown, container))
+            _audio.PlayPvs(ent.Comp.InsertSound, ent);
+        else
+            throw new InvalidOperationException("Container insertion failed but CanInsert returned true");
```
Comment (deltanedas) [inline Content.Server/Containers/ThrowInsertContainerSystem.cs:48]: if (!Insert()) throw() better

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by TheShuEd** [comment]
instant aftermerg review

**Comment by metalgearsloth** [comment]
The prob one needs doing the rest idc.

**Pull request merged**